hera_qm/ant_metrics.py

```diff
@@ -431,3 +436,155 @@ def save_antenna_metrics(self, metricsJSONFilename):
+                fn = re.sub(file_pol, pol, filename)
+                full_filename = os.path.join(dirname, fn)
+                if not os.path.exists(full_filename):
+                    warnings.warn("Could not find " + full_filename + "; skipping that JD")
+                    pols_exist = False
+                    break
+            if pols_exist:
+                # add all pols to file_list
+                for pol in pol_list:
+                    fn = re.sub(file_pol, pol, filename)
+                    full_filename = os.path.join(dirname, fn)
+                    file_list.append(full_filename)
+
+    return file_list
+
+def ant_metrics_run(files, opts):
+    """
+    Run a series of ant_metrics tests on a given set of input files.
+
+    Args:
+       files -- a list of files to run ant metrics on. Can be any of the 4 polarizations
+       opts -- an optparse OptionParser instance
+    Return:
+       None
+
+    The funciton will take in a list of files and options. It will run the
+    sereis of ant metrics tests, and produce a JSON file containing the relevant
+    information. The file list passed in need only contain one of the polarization
+    files for a given JD, and the function will look for the other polarizations
+    in the same folder. If not all four polarizations are found, a warning is
+    generated, since the code assumes all four polarizations are present.
+    """
+    # XXX: does this create a circular dependency if we only do this inside the function?
+    try:
+        from hera_cal import omni
+    except(ImportError):
+        from nose.plugins.skip import SkipTest
+        raise SkipTest('hera_cal.omni not detected. It must be installed to calculate array info')
+
+    # define polarizations to look for
+    # XXX: This will have to change if file naming convention changes from 'xx', 'xy', etc.
+    pol_list = ['xx', 'xy', 'yx', 'yy']
+
+    # check that we were given some files to process
+    if len(files) == 0:
+        raise AssertionError('Please provide a list of visibility files')
+
+    # generate a list of all files to be read in
+    fullpol_file_list = generate_fullpol_file_list(files, pol_list)
+    if len(fullpol_file_list) == 0:
+        raise AssertionError('Could not find all 4 polarizations for any files provided')
+
+    # define freqs
+    # XXX: might want to make this flexible in the future
+    freqs = np.linspace(0.1, 0.2, num=1024, endpoint=False)
+
+    # process calfile
+    aa = aipy.cal.get_aa(opts.cal, freqs)
+    info = omni.aa_to_info(aa, pols=[pol_list[-1][0]], crosspols=[pol_list[-1]])
+    reds = info.get_reds()
+
+    # do the work
+    num_files = len(fullpol_file_list) // 4
+    for i in range(num_files):
+        file_list = fullpol_file_list[4*i:4*(i+1)]
+        am = Antenna_Metrics(file_list, reds, fileformat='miriad')
+        am.iterative_antenna_metrics_and_flagging(crossCut=opts.crossCut, deadCut=opts.deadCut, verbose=True)
```

Comment:
I actually made a `-q` flag for quiet operation, i.e., the default is to be verbose.